src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -954,7 +955,27 @@ public bool RevisionIsLast(DocumentsOperationContext context, Table table, Slice
             return true;
         }
 
-        private void DeleteRevisionFromTable(DocumentsOperationContext context, Table table, Dictionary<string, Table> writeTables,
+        public bool ShouldAdoptRevision(DocumentsOperationContext context, Slice lowerId, Slice lowerIdPrefix, CollectionName collectionName, out Table table, out Document lastRevision)
+        {
+            lastRevision = null;
+            table = null;
+
+            var local = _documentsStorage.GetDocumentOrTombstone(context, lowerId, throwOnConflict: false);
+            if (local.Document != null) // doc isn't deleted, so we don't need to create delete revision
+                return false;
+
+            using (GetKeyWithEtag(context, lowerId, etag: long.MaxValue, out var compoundPrefix))
+            {
+                table = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, collectionName);
+                var holder = table.SeekOneBackwardFrom(RevisionsSchema.Indexes[IdAndEtagSlice], lowerIdPrefix, compoundPrefix);
+                if (holder == null) // revisions was already deleted by now so we don't need to add delete revision
```

Comment:
the SeekOneBackwardFrom returns null when it doesn't find anything.
It happens in case the revisions were already deleted by now so we don't need to add delete revision.